File: codeWars/Level_6/camelCase.js
/** Camel Case
 * Complete the solution so that the function will break up camel casing, 
 * using a space between words.
 * 
 * "camelCasing"  =>  "camel Casing"
 * "identifier"   =>  "identifier"
 * ""             =>  ""
 */

// complete the function
function solution(string) {
    let outputString = "";
    for(let i = 0; i < string.length; i++) {
        if (string[i] === string[i].toUpperCase() ) {
            outputString = outputString + " ";
        }
        outputString = outputString + string[i];
    }
    return outputString;
}


const res1 = solution("camelCasing");
const res2 = solution("identifier");
const res3 = solution("MyNameIs");
const res4 = solution("youAndMe");

console.log(`Result 1: ${res1}\nResult 2: ${res2}\nResult 3: ${res3}\nResult 4: ${res4}`);

File: codeWars/Level_6/camelCase.py
# Camel Case
# Complete the solution so that the function will break up camel casing, 
# using a space between words.
 
# "camelCasing"  =>  "camel Casing"
# "identifier"   =>  "identifier"
# ""             =>  ""


# Complete the function
def camelCase(string):
    outputString = "" + string[0].upper()
    for char in range(1, len(string)):
        upperCase = string[char].upper()
        if (string[char] == upperCase):
            outputString = outputString + " "
        outputString = outputString + string[char].lower()
    return outputString + "."


# Tests
tests = ["camelCasing", "astronaut", "yourNameIs", "weAreTheChampions"]
for test in range(len(tests)):
    print("Result {0}: {1}".format(test+1, camelCase(tests[test])))
